src/Frontend/Components/DiffPopup/DiffPopup.tsx

```diff
@@ -20,31 +27,86 @@ interface DiffPopupProps {
 
 export function DiffPopup({
   current,
-  isOpen,
   original,
+  isOpen,
   setOpen,
 }: DiffPopupProps) {
-  const [originalFormConfig, currentFormConfig] = useAttributionFormConfigs({
-    current,
+  const dispatch = useAppDispatch();
+  const {
+    originalFormConfig,
+    bufferFormConfig,
+    bufferPackageInfo,
+    setBufferPackageInfo,
+  } = useAttributionFormConfigs({
     original,
+    current,
+    isOpen,
   });
 
+  function checkIfWasPreferredAndLicenseInfoResetNecessaryAndApply(
+    original: PackageInfo,
+    current: PackageInfo,
+    buffer: PackageInfo,
+  ) {
+    if (
+      isEqual(
+        getComparableAttributes(buffer),
+        getComparableAttributes(original),
+      ) &&
+      original.wasPreferred
+    ) {
+      buffer.wasPreferred = true;
+    }
+    if (current.firstParty && buffer.firstParty) {
+      buffer.copyright = current.copyright;
+      buffer.licenseName = current.licenseName;
+      buffer.licenseText = current.licenseText;
+    }
+    dispatch(setTemporaryDisplayPackageInfo(buffer));
+  }
+
   return (
     <NotificationPopup
       header={text.diffPopup.title}
       content={renderDiffView()}
       leftButtonConfig={{
-        disabled: true,
+        disabled: isEqual(
+          getComparableAttributes(bufferPackageInfo),
+          getComparableAttributes(current),
+        ),
         buttonText: text.buttons.diffPopup.applyChanges,
+        onClick: () => {
+          checkIfWasPreferredAndLicenseInfoResetNecessaryAndApply(
+            original,
+            current,
+            bufferPackageInfo,
+          );
+          setOpen(false);
+        },
       }}
       centerRightButtonConfig={{
-        disabled: true,
+        disabled: isEqual(
+          getComparableAttributes(bufferPackageInfo),
+          getComparableAttributes(original),
+        ),
         buttonText: text.buttons.diffPopup.revertAll,
+        onClick: () => {
+          setBufferPackageInfo({
+            ...FORM_ATTRIBUTES.reduce(
+              (buffer, attribute) => {
+                // @ts-expect-error original[attribute] will always have a value that is allowed for buffer[attribute] but typescript does not recognize that
+                buffer[attribute] = original[attribute];
+                return buffer;
+              },
+              { ...bufferPackageInfo },
+            ),
+          });
+        },
```

Comment:
this will do the job without type errors or accidentally mutating reactive state:
```suggestion
        onClick: () => {
          setBufferPackageInfo({
            ...bufferPackageInfo,
            ...FORM_ATTRIBUTES.reduce(
              (acc, attribute) => ({
                ...acc,
                [attribute]: original[attribute],
              }),
              {},
            ),
          });
        },
```

Reply:
Nice.